integr8ly/integreatly-operator
# MGDAPI-6351 - custom keycloak image, add extentions

# Issue link
Jira: https://issues.redhat.com/browse/MGDAPI-6351

## This is Investigation PR, not for merge

# What
Investigate creation and use of our custom keycloak image

* Custom RHBK Image. authdelay.jar plugin added
*  authdelay.jar build from this project: https://github.com/marcelomrwin/authentication-delay-plugin
     $mvn clean package 

# Verification steps
- create Custom RHBK  image:
```
make custom-rhbk 
```
- create Custom CR, and define image there, as in example below:
 (please you can look sample in Jira)
- work according to Migration doc (see Jira)
- After RHBK Installation (pods are running) 
  - Login to RHBK Admin portal using edge route UDL (+ /auth)
  - Master Realm / Provider Info / authenticator  --  and see that `delay-authentication` is present in providers list
  - You also can login to KC pod and see that authdelay.jar is located at /opt/keycloak/providers/authdelay.jar
  - Login to 3scale - via IDP using new user, and see that Authentication delay message is present on screen
